### PR TITLE
VAS-153 Lyra BLE connection DAPI

### PR DESCRIPTION
**Pull request by JulianASLaird**
Initial commit of Lyra connection DAPI.
Requires connection pool code as no permanent storage of the connection is exposed to by the SDK other than the handle and certain information is only available for the lifetime of the callback function associated with that BLE event.
The connection pool manages storage of this event data e.g. address of connecting device, internal handle,  role, etc.

### CODE REVIEW

File: include/pi_dapi_ble_connection.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+ * @brief Platform Interface XXXX-level API
```
Comment (greg-leach) [inline include/pi_dapi_ble_connection.h:2]: DAPI?

File: include/pi_dapi_ble_connection_pool.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+ */
+#ifndef PLATFORM_INTERFACE_DAPI_PI_DAPI_LYRA_BLE_CONNECTION_INCLUDE_PI_DAPI_BLE_CONNECTION_POOL_H_
+#define PLATFORM_INTERFACE_DAPI_PI_DAPI_LYRA_BLE_CONNECTION_INCLUDE_PI_DAPI_BLE_CONNECTION_POOL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**************************************************************************************************/
+/* Includes																						  */
```
Comment (greg-leach) [inline include/pi_dapi_ble_connection_pool.h:16]: Formatting goes funny here.

File: include/pi_dapi_ble_connection_pool.h [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	sl_bt_evt_connection_phy_status_t phy;
+	sl_bt_evt_connection_rssi_t rssi;
+	sl_bt_evt_gatt_mtu_exchanged_t mtu;
+} PI_DAPI_CONNECTION_T;
+
+/**************************************************************************************************/
+/* Global Function Definitions                                                                    */
+/**************************************************************************************************/
+/*!
+ * @brief				Find an entry in the connection pool with a matching connection handle
```
Comment (greg-leach) [inline include/pi_dapi_ble_connection_pool.h:49]: Spacing goes awry again.

File: src/pi_dapi_ble_connection.c [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+#define CONN_RESPONDER_LATENCY 0
+#define CONN_MIN_CE_LENGTH 0
+#define CONN_MAX_CE_LENGTH 0xffff
+
+#define CONN_INTERVAL_FROM_MS(x) (((x)*4) / 5)
+#define CONN_EVENT_FROM_MS(x) (((x)*5) / 8)
+#define TIMEOUT_FROM_MS(x) ((x) / 10)
+
+/**************************************************************************************************/
+/* Local Data Storage		                                                                      */
```
Comment (greg-leach) [inline src/pi_dapi_ble_connection.c:46]: Maybe here too.

**Review by greg-leach — APPROVED** [review]

**Review by randyscott — APPROVED** [review]

File: src/pi_dapi_ble_connection.c [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+	/* Calculate the connection parameters */
+	uint16_t min_conn_interval =
+		CONN_INTERVAL_FROM_MS(pi_shared_ble_connection.min_conn_interval_ms);
+	uint16_t max_conn_interval =
+		CONN_INTERVAL_FROM_MS(pi_shared_ble_connection.max_conn_interval_ms);
+	uint16_t timeout = TIMEOUT_FROM_MS(pi_shared_ble_connection.timeout_ms);
+	uint8_t phy_to_use = sl_bt_gap_phy_1m;
+
+	/* Use coded PHY? */
+	if ((phy & PI_HAPI_BLE_COMMON_PHY_CODED) != 0) {
```
[the remark this answers is not in the document]
Comment (JulianASLaird) [reply src/pi_dapi_ble_connection.c:241]: No only 1m or coded are available at this point...

@param[in] initiating_phy Enum @ref sl_bt_gap_phy_t. The initiating PHY.
 *   Values:
 *     - <b>sl_bt_gap_phy_1m (0x1):</b> 1M PHY
 *     - <b>sl_bt_gap_phy_coded (0x4):</b> Coded PHY, 125k (S=8) or 500k (S=2)

File: include/pi_dapi_ble_connection.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+#define PLATFORM_INTERFACE_DAPI_PI_DAPI_LYRA_BLE_CONNECTION_INCLUDE_PI_DAPI_BLE_CONNECTION_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**************************************************************************************************/
+/* Local Constants, Macros and Type Definitions                                                   */
+/**************************************************************************************************/
+#define MAX_CONN 16
```
[the remark this answers is not in the document]
Comment (JulianASLaird) [reply include/pi_dapi_ble_connection.h:18]: Yes

**Pull request merged**